NPM Package Fails to Install on Mingw

## 🐛 Bug description
The npm package fails to install on mingw:
```
npm ERR!   throw new Error(`Unsupported platform: ${type} ${arch}`);
npm ERR!   ^
npm ERR!
npm ERR! Error: Unsupported platform: MINGW32_NT-10.0 x64
```

This is due to the script not detecting the Windows environment correctly. I have a branch at https://github.com/nathaniel-daniel/wasm-pack/tree/mingw that fixes the issue, which I can turn into a PR.

#### 🤔 Expected Behavior
The package should have installed.

#### 👟 Steps to reproduce
1. Install msys2.
2. Enter the ucrt64 env (`ucrt64`)
3. Install nodejs (`pacman -S ucrt64/mingw-w64-ucrt-x86_64-nodejs`)
4. Create a new project with `wasm-pack` as a dependency or dev dependency.
5. Run `npm i`

#### 🌍 Your environment
Include the relevant details of your environment.
wasm-pack version: N/A (Issue with NPM package)
rustc version: N/A (Issue with NPM package)

--- a/npm/binary.js
+++ b/npm/binary.js
@@ -7,7 +7,8 @@ const getPlatform = () => {
   const type = os.type();
   const arch = os.arch();
 
-  if (type === "Windows_NT" && arch === "x64") {
+  // https://github.com/nodejs/node/blob/c3664227a83cf009e9a2e1ddeadbd09c14ae466f/deps/uv/src/win/util.c#L1566-L1573
+  if ((type === "Windows_NT" || type.startsWith("MINGW32_NT-")) && arch === "x64") {
     return windows;
   }
   if (type === "Linux" && arch === "x64") {

